chore: migrate to `typescript-eslint` and `airbnb` style guide

barrett-schonefeld opened the pull request:
## PR summary
Migrate the TypeScript linter from `TSLint` to `typescript-eslint` because `TSLint` is deprecated.

Also utilizes the `airbnb` style guidelines and the `jest` linting plugin.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).

## Current vs new behavior  
Uses a new `eslint` config that utilizes `typescript-eslint`, `jest` linting, and the `airbnb` style guidelines.

## Does this PR introduce a breaking change?    
No.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/IBM/platform-services-node-sdk?pullRequest=103) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/IBM/platform-services-node-sdk?pullRequest=103) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/IBM/platform-services-node-sdk?pullRequest=103) it.</sub>

padamstx commented:
@barrett-schonefeld @dpopp07 looks like the `npm run check-packages` step failed in the travis builds.   Not even sure what would cause that... unfortunately, the log doesn't include any details, just that the command failed.

barrett-schonefeld commented:
> I'm not sure what might have happened. Is it possible that "installed-check" was being installed for us because it was a dependency of "tslint" which we no longer depend on???

It was a bit simpler. I removed `installed-check` on accident when I added the `typescript-eslint` dependencies to `package.json`.

I added the dependency back and pushed the changes.

dpopp07 reviewed (approved):
Changes all look consistent with changes in the other files with the new style. Looks good 👍

padamstx reviewed (approved):
LGTM

barrett-schonefeld commented:
Changes look good!

ibm-devx-sdk commented:
:tada: This PR is included in version 0.18.9 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/ibm-platform-services/v/0.18.9)
- [GitHub release](https://github.com/IBM/platform-services-node-sdk/releases/tag/v0.18.9)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
